src/gluonts/dataset/pandas.py

```diff
@@ -171,31 +169,30 @@ def _pair_to_dataentry(self, pair) -> DataEntry:
 
         entry = {
             "start": df.index[0],
-            "target": remove_last_n(
-                self.ignore_last_n_targets,
-                df[self.target].values.transpose(),
-            ),
         }
 
+        target = df[self.target].values
+        target = target[: len(target) - self.future_length]
+        entry["target"] = target.T
+
         if item_id is not None:
             entry["item_id"] = item_id
 
-        if self.num_feat_static_cat > 0:
-            entry["feat_static_cat"] = self._static_cats.loc[item_id].values
+        if self.num_feat_static_cat:
```

Comment:
I prefer `self.num_feat_static_cat > 0` (not a fan of implicit conversions to `bool`)

Reply:
But that's the pythonic way to do it.